src/ucp/core/ucp_mm.h

```diff
@@ -24,6 +24,15 @@
  * The array itself contains only the MDs specified in md_map, without gaps.
  */
 typedef struct ucp_rkey {
+    /* cached values for the most recent endpoint configuration */
+    struct {
+        ucp_ep_cfg_index_t        ep_cfg_index; /* EP configuration relevant for the cache */
+        ucp_lane_index_t          rma_lane;     /* Lane to use for RMAs */
+        ucp_lane_index_t          amo_lane;     /* Lane to use for AMOs */
+        unsigned                  max_put_short;/* Cached value of max_put_short */
+        uct_rkey_t                rma_rkey;     /* Key to use for RMAs */
+        uct_rkey_t                amo_rkey;     /* Key to use for AMOs */
+    } c;
```

Comment:
'c' - to make it shorter i all places it's used
bcopy_threshold - not cached to avoid enlarging rkey structure (potential cache misses), since max_short is fast-path and 32bit it's the only one worth caching IMHO